### PR TITLE
fix: Unsigned type related bugs

### PR DESCRIPTION
**Pull request by kazuyukitanimura**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion-comet/issues/1067

## Rationale for this change

Bug fix. A few expressions were failing some unsigned type related tests

## What changes are included in this PR?

 - For `u8`/`u16`, switched to use `generate_cast_to_signed!` in order to copy full i16/i32 width instead of padding zeros in the higher bits
 - `u64` becomes `Decimal(20, 0)` but there was a bug in `round()`  (`>` vs `>=`)

## How are these changes tested?

Put back tests for unsigned types

### CODE REVIEW

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @kazuyukitanimura

File: native/core/src/parquet/read/values.rs [hunk 1]
```diff
@@ -586,8 +586,6 @@ macro_rules! generate_cast_to_unsigned {
     };
 }
 
-generate_cast_to_unsigned!(copy_i32_to_u8, i32, u8, 0_u8);
-generate_cast_to_unsigned!(copy_i32_to_u16, i32, u16, 0_u16);
 generate_cast_to_unsigned!(copy_i32_to_u32, i32, u32, 0_u32);
```
Comment (viirya) [inline native/core/src/parquet/read/values.rs:589]: Same logic doesn't apply for u32?

Comment (kazuyukitanimura) [reply native/core/src/parquet/read/values.rs:589]: `copy_i32_to_u32` stays the same as it is for widening `i32` -> `i64(u32)`, so we need to pad zeros

File: spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala [hunk 2]
```diff
@@ -434,8 +434,8 @@ abstract class ParquetReadSuite extends CometTestBase {
                   i.toFloat,
                   i.toDouble,
                   i.toString * 48,
-                  java.lang.Byte.toUnsignedInt((-i).toByte),
-                  java.lang.Short.toUnsignedInt((-i).toShort),
```
Comment (viirya) [inline spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:438]: Do you check if Spark also return these values?

Comment (kazuyukitanimura) [reply spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:438]: I would say that is covered by the `"basic data type support"` test

Comment (viirya) [reply spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:438]: I meant that you changed this expected results to match what Comet returns, but are they same as Spark returns?

Comment (kazuyukitanimura) [reply spark/src/test/scala/org/apache/comet/parquet/ParquetReadSuite.scala:438]: Yes the same as Spark

**Pull request merged**